src/xo-app/host/tab-network.js

```diff
@@ -89,10 +90,64 @@ class ConfigureIpModal extends Component {
 @connectStore(() => ({
   vifsByNetwork: createGetObjectsOfType('VIF').groupBy('$network')
 }))
-class PifItem extends Component {
+class PifItemVlan extends Component {
+  _editPif = vlan =>
+    editPif(this.props.pif, { vlan })
+  render () {
+    const { pif } = this.props
+    return <div>{pif.vlan === -1
+      ? 'None'
+      : <Number value={pif.vlan} onChange={this._editPif}>
+        {pif.vlan}
+      </Number>
+    }</div>
+  }
+}
+
+class PifItemIp extends Component {
   state = { configModes: [] }
 
-  componentWillMount () {
+  componentDidMount () {
+    getIpv4ConfigModes().then(configModes =>
+      this.setState({ configModes })
+    )
+  }
+
+  _configIp = () => {
+    return confirm({
+      icon: 'ip',
+      title: _('pifConfigureIp'),
+      body: <ConfigureIpModal pif={this.props.pif} />
+    }).then(
+      params => {
+        if (!params.ip || !params.netmask) {
+          error(_('configIpErrorTitle'), _('configIpErrorMessage'))
+          return
+        }
+        return reconfigurePifIp(this.props.pif, { mode: 'Static', ...params })
+      },
+      noop
+    )
```

Comment:
If I'm not mistaken, this whole part is duplicated. If that's the case, create a function outside the class that does this.
```js
const reconfigureStaticIp = pif => {
  // ...
}
```